canonical/mir
# feature: implementation of xdg_activation_v1 because I want my auto focuses

[Screencast from 2024-10-16 16-21-05.webm](https://github.com/user-attachments/assets/86f9b0bf-181c-4aab-acfd-4ceb1e4e2c38)


## What's new?
- Added the xdg_activation_v1 protocol
- Implemented the xdg_activation_v1 protocol

## To test
- Run `miral-shell`
- Open `gnome-terminal`
- Run `gedit`
- Focus `gnome-terminal` again
- Run `gedit` again
- Note that `gedit` gets focus :magic_wand: 

Similarly:
- Open up `emacs`
- Open up `firefox`
- In emacs, write `https://google.com`
- Ctrl + Click the link
- Watch as firefox gets focus!

## Open questions
- The "timeout" is not configurable at the moment, but perhaps it should be. However, I don't think we do per-protocol options AFAICT

## Todos
- I should probably write some WLCS tests!
